doc/languages-frameworks/gnome.xml

```diff
@@ -0,0 +1,211 @@
+ <title>GNOME</title>
+
+ <section xml:id="ssec-gnome-packaging">
+  <title>Packaging GNOME applications</title>
+
+  <para>
+   Programs in the GNOME universe are written in various languages but they all
+   use GObject-based libraries like GLib, GTK+ or GStreamer. These libraries
+   are often modular, relying on looking into certain directories to find their
+   modules. However, due to Nix’s specific file system organization, this
+   will fail without our intervention. Fortunately, the libraries usually allow
+   overriding the directories through environment variables, either natively or
+   thanks to a patch in nixpkgs.
+   <link xlink:href="#fun-wrapProgram">Wrapping</link> the executables to
+   ensure correct paths are available to the application constitutes a
+   significant part of packaging a modern desktop application. In this section,
+   we will describe various modules needed by such applications, environment
+   variables needed to make the modules load, and finally a script that will do
+   the work for us.
+  </para>
+
+  <section xml:id="ssec-gnome-settings">
+   <title>Settings</title>
+
+   <para>
+    <link xlink:href="https://developer.gnome.org/gio/stable/GSettings.html">GSettings</link>
+    API is often used for storing settings. GSettings schemas are required, to
+    know the type and other metadata of the stored values. GLib looks for
+    <filename>glib-2.0/schemas/gschemas.compiled</filename> files inside the
+    directories of <envar>XDG_DATA_DIRS</envar>.
+   </para>
+
+   <para>
+    On Linux, GSettings API is implemented using
+    <link xlink:href="https://wiki.gnome.org/Projects/dconf">dconf</link>
+    backend. You will need to add <literal>dconf</literal> GIO module to
+    <envar>GIO_EXTRA_MODULES</envar> variable, otherwise the
+    <literal>memory</literal> backend will be used and the saved settings will
+    not be persistent.
+   </para>
+
+   <para>
+    Last you will need the dconf database itself. Installing
+    <package>gnome3.dconf</package> to
+    <option>environment.systemPackages</option> will be enough.
+   </para>
+
+   <para>
+    Some applications will also require
+    <package>gsettings-desktop-schemas</package> for things like reading proxy
+    configuration or user interface customization. This dependency is often not
+    mentioned by upstream, you should grep for
+    <literal>org.gnome.desktop</literal> and
+    <literal>org.gnome.system</literal> to see if the schemas are needed.
+   </para>
+  </section>
+
+  <section xml:id="ssec-gnome-icons">
+   <title>Icons</title>
+
+   <para>
+    When an application uses icons, an icon theme should be available in
+    <envar>XDG_DATA_DIRS</envar>. The default
+    <link xlink:href="https://www.freedesktop.org/wiki/Software/icon-theme/">hicolor-icon-theme</link>
+    is recommended, unless the application needs some specific icons not
+    available in there.
```

Comment:
Yeah, this is nonsense. Anyone know why are we adding this other than for clean-up hook?

Reply:
It seems people add it when they're running under systems missing icon themes, and they see
* https://gitlab.gnome.org/GNOME/gtk/blob/master/gtk/gtkicontheme.c#L1861

Reply:
And I guess XDG_ICON_DIRS